Identifier.ts

```diff
@@ -25,14 +25,33 @@ export namespace Identifier {
 	export function toUint48(identifier: Identifier): number {
 		return Number.parseInt(toHexadecimal(identifier, 12), 16)
 	}
-	export function fromBinary(identifier: Uint8Array): Identifier {
-		return Base64.encode(identifier, "url")
+	export function fromBinary(identifier: Uint8Array, standard: Base64.Standard = "url"): Identifier {
+		return Base64.encode(identifier, standard)
 	}
 	export function toBinary(identifier: Identifier): Uint8Array {
 		return Base64.decode(identifier, "url")
 	}
-	export function generate(length: Length): Identifier {
-		return fromBinary(crypto.getRandomValues(new Uint8Array((length / 4) * 3)))
+	export function generate(length: Length): Identifier
+	export function generate(
+		length: Length,
+		ordering: Extract<Base64.Standard, "ordered" | "reversed">,
+		value?: number | Uint8Array | string
+	): Identifier
+	export function generate(
+		length: Length,
+		ordering?: Extract<Base64.Standard, "ordered" | "reversed">,
+		value?: number | Uint8Array | string
```

Comment:
when having multiple optionns to a function. do we prefer this style of parameters? using double optional parameters can force you to pass an unessesary undefined to the call.

```ts
function foo(required: string, {a = 123, b = 456}: {a?: number, b?: number} = {}) {
    console.log(required, a == 123 ? "default a" : "non default a", b == 456 ? "default b" : "non default b")
}
foo("call 1")
foo("call 2", {a: 321})
foo("call 2", {b: 654})
foo("call 4", {a: 321, b: 654})
```
https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUAnApgRxDbAEwC5EBnKDGMAcwBpEBvAQ0QF5EBGAJgGYGARu0QAWAKwA2AL6kWAflJgQAWwFYMghYiWr1U4YykBKJgChEFxBARk4AGywA6O3BqZc+Ig1ZsOPXohyiABEhFjAzCB2UIjMwYikwWAIiGERUTFxguwc4hKBIWmR0YgC8YnJSEUZpcFGplKmKOjBEMx2dlx1Tahore2d3MEMLKS83JzGPS1tHYhDIwKkEmIiU819s50iw0zMYxOCy6vGQA

Reply:
Marcus wanted it this way.. I agree with you, we'll see what he says since he often changes his mind about these things.

Reply:
Changing the order of these parameters affect the need for `undefined` in the parameter. If you're not using a `value` why would you ever use ordering? There are a couple of options, either we order them as `length, value?, ordering?` with a default if ordering is undefined or we make an overload where both are required.

Reply:
Both are now required in overload.